lib/parser/architecture-parser.ts

```diff
@@ -26,6 +26,20 @@ export class ArchitectureParser extends ParserBase {
       this.architecture = new OArchitecture(this.parent, this.getToken().range.copyExtendEndOfLine());
     } else if (structureName === 'block') {
       this.architecture = new OBlock(this.parent, this.getToken().range.copyExtendEndOfLine());
+      // guarded block
+      if (this.getToken().getLText() === '(') {
+        const startRange = this.getToken().range;
+        this.consumeToken(); // consume '('
+        (this.architecture as OBlock).guardCondition = this.extractReads(this.architecture, this.advanceClosingParenthese());
+        const guardRange = startRange.copyWithNewEnd(this.getToken().range.end);
+        // implicit declare constant GUARD
+        const constant = new OConstant(this.architecture, guardRange);
+        constant.lexerToken = new OLexerToken('GUARD', guardRange, TokenType.basicIdentifier);
```

Comment:
guard is a implicit signal according to the VHDL standard, any specific reason why this is a constant?

Reply:
Ah i see it is in the warning.
The proper way would probably be let a signal be implicit, that is not checked than, (constant not giving not read warnings seems to be a bug to me tbh...)

Reply:
It is a constant because it cannot be assigned, only read. And to avoid the not read warning an extra read is added in the lines below (with the explaining comment). This is a bit hacky but I find it the best and easiest solution.

Reply:
Okay, fine i guess ;)